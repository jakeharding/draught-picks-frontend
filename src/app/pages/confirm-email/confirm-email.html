<!--
  Generated template for the ConfirmEmailPage page.

  See http://ionicframework.com/docs/components/#navigation for more info on
  Ionic pages and navigation.
-->
<ion-header>
  <draught-header></draught-header>
</ion-header>


<ion-content class="ion-padding">
  <ion-row>
    <ion-col
        class="below-header ion-text-justify"
        size="12" sizeSm="10" offsetSm="1" sizeMd="8" offsetMd="2" sizeLg="6" offsetLg="3" sizeXl="4" offsetXl="4"
    >
      <div class="ion-text-center">
        <p>{{ thankYouMessage }}</p>
        <p>{{ clickMessage }}</p>
        <a class="pointer ion-text-center" (click)="goToLink()">{{ linkText }}</a>
      </div>
    </ion-col>
  </ion-row>

</ion-content>
