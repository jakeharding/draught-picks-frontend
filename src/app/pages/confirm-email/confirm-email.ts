import { Component } from '@angular/core';
import {  NavController } from '@ionic/angular';
import { BasePage } from '../BasePage';
import { UserProvider } from '../../services/user/user';
import { ToastProvider } from '../../services/toast/toast';
import { ActivatedRoute } from '@angular/router';

/**
 * Generated class for the ConfirmEmailPage page.
 *
 * See https://ionicframework.com/docs/components/#navigation for more info on
 * Ionic pages and navigation.
 */

@Component({
  selector: 'page-confirm-email',
  templateUrl: 'confirm-email.html',
  styleUrls: ['confirm-email.scss']
})
export class ConfirmEmailPage extends BasePage {
  showResendLink = false;
  linkText = 'Sign in.';
  thankYouMessage = 'Thank you for confirming your email!';
  clickMessage = 'Please click the link below to sign in.';
  confirmKey: string;
  constructor(public navCtrl: NavController,
              private route: ActivatedRoute,
              private userProvider: UserProvider,
              private toastProvider: ToastProvider) {
    super(`confirm-email/${route.snapshot.paramMap.get('key')}`);
    this.confirmKey = route.snapshot.paramMap.get('key');
  }

  ionViewDidEnter() {
      this.userProvider.confirmEmail({key: this.confirmKey}).catch(err => {
        this.showResendLink = this.isClientError(err.status);
        if (this.showResendLink) {
          this.thankYouMessage = err.error.confirm_key;
          this.clickMessage = 'Click the link below to resend the confirmation email.';
          this.linkText = 'Resend confirmation email';
        } else {
          this.toastProvider.errorToast();
          this.thankYouMessage = ToastProvider.defaultErrorMsg;
        }
        return err;
      });
  }

  goToLink() {
    let link = 'sign-in';
    if (this.showResendLink) {
      link = 'resend-email';
    }
    this.navCtrl.navigateRoot(link);
  }
}
