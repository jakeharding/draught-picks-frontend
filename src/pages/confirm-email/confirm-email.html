<!--
  Generated template for the ConfirmEmailPage page.

  See http://ionicframework.com/docs/components/#navigation for more info on
  Ionic pages and navigation.
-->
<draught-header></draught-header>


<ion-content padding>
  <ion-row>
    <ion-col class="below-header" text-justify size="20px" col-12 col-sm-10 offset-sm-1 col-md-6 offset-md-3 col-lg-6 offset-lg-3 col-xl-4 offset-xl-4>
      <p>{{ thankYouMessage }}</p>
      <p>{{ clickMessage }}</p>
      <div text-center>
        <a text-center class="underline" (click)="goToLink()">{{ linkText }}</a>
      </div>
    </ion-col>
  </ion-row>

</ion-content>