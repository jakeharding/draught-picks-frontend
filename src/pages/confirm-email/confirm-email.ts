import { Component } from '@angular/core';
import { IonicPage, NavController, NavParams } from 'ionic-angular';
import { BasePage } from '../BasePage';
import { SignInPage } from '../sign-in/sign-in';
import { UserProvider } from '../../providers/user/user';
import { ResendEmailPage } from '../resend-email/resend-email';
import { Page } from 'ionic-angular/umd/navigation/nav-util';
import { ToastProvider } from '../../providers/toast/toast';

/**
 * Generated class for the ConfirmEmailPage page.
 *
 * See https://ionicframework.com/docs/components/#navigation for more info on
 * Ionic pages and navigation.
 */

@IonicPage({
  name: 'confirm-email',
  segment: 'confirm-email/:key'
})
@Component({
  selector: 'page-confirm-email',
  templateUrl: 'confirm-email.html',
})
export class ConfirmEmailPage extends BasePage {
  showResendLink: boolean = false;
  linkText: string = 'Sign in.';
  thankYouMessage: string = 'Thank you for confirming your email!';
  clickMessage: string = 'Please click the link below to sign in.';
  constructor(public navCtrl: NavController,
              private navParams: NavParams,
              private userProvider: UserProvider,
              private toastProvider: ToastProvider) {
    super(`confirm-email/${navParams.get('key')}`);
  }

  ionViewDidEnter() {
      this.userProvider.confirmEmail(this.navParams.data).catch(err => {
        this.showResendLink = this.isClientError(err.status);
        if (this.showResendLink) {
          this.thankYouMessage = err.error.confirm_key;
          this.clickMessage = 'Click the link below to resend the confirmation email.';
          this.linkText = 'Resend confirmation email';
        } else {
          this.toastProvider.errorToast();
          this.thankYouMessage = ToastProvider.defaultErrorMsg;
        }
        return err;
      });
  }

  goToLink() {
    let link: Page = SignInPage;
    if (this.showResendLink) {
      link = ResendEmailPage;
    }
    this.navCtrl.setRoot(link);
  }
}
