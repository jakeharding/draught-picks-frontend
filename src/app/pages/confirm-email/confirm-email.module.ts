import { NgModule } from '@angular/core';
import { IonicModule } from '@ionic/angular';
import { ConfirmEmailPage } from './confirm-email';
import { RouterModule } from '@angular/router';
import { ComponentsModule } from '../../components/components.module';

@NgModule({
  declarations: [
    ConfirmEmailPage,
  ],
  imports: [
    IonicModule,
    RouterModule.forChild([{path: '', component: ConfirmEmailPage}]),
    ComponentsModule,
  ],
})
export class ConfirmEmailPageModule {}
