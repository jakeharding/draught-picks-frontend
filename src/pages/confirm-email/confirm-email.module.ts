import { NgModule } from '@angular/core';
import { IonicPageModule } from 'ionic-angular';
import { ConfirmEmailPage } from './confirm-email';
import { ComponentsModule } from '../../components/components.module';

@NgModule({
  declarations: [
    ConfirmEmailPage,
  ],
  imports: [
    IonicPageModule.forChild(ConfirmEmailPage),
    ComponentsModule
  ],
})
export class ConfirmEmailPageModule {}
