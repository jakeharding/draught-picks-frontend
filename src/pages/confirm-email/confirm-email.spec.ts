/**
 * confirm-email.spec.ts
 *
 * Created by jake
 * Created on 10/21/18
 */

import {async, ComponentFixture, TestBed} from "@angular/core/testing";
import { ConfirmEmailPage } from "./confirm-email";
import {NO_ERRORS_SCHEMA} from "@angular/core";
import {NavController} from "ionic-angular";

describe('ConfirmEmailPage', () => {

  let fixture: ComponentFixture<ConfirmEmailPage>;
  let instance: ConfirmEmailPage;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [NO_ERRORS_SCHEMA],
      declarations: [ConfirmEmailPage],
      providers: [
        { provide: NavController, useValue: {}}
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ConfirmEmailPage);
    instance = fixture.debugElement.componentInstance;
  });

  it('should initialize the ConfirmEmailPage', () => {
    expect(instance).toBeTruthy();
  })
});