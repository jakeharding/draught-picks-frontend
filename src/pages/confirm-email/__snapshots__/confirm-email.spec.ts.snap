// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmEmailPage ionViewWillEnter on failed promise with client error 1`] = `
<page-confirm-email
  clickMessage={[Function String]}
  linkText={[Function String]}
  navCtrl={[Function Object]}
  navParams={[Function NavParams]}
  thankYouMessage={[Function String]}
  toastProvider={[Function Object]}
  userProvider={[Function Object]}
>
  
  <draught-header />
  
  
  <ion-content
    padding=""
  >
    
    
    <ion-row>
      
      
      <ion-col
        class="below-header"
        col-12=""
        col-lg-6=""
        col-md-6=""
        col-sm-10=""
        col-xl-4=""
        offset-lg-3=""
        offset-md-3=""
        offset-sm-1=""
        offset-xl-4=""
        size="20px"
        text-justify=""
      >
        
        
        <p>
          Thank you for confirming your email!
        </p>
        
        
        <p>
          Please click the link below to sign in.
        </p>
        
        
        <div
          text-center=""
        >
          
          
          <a
            class="underline"
            text-center=""
          >
            Sign in.
          </a>
          
        
        </div>
        
      
      </ion-col>
      
    
    </ion-row>
    
  
  
  </ion-content>
</page-confirm-email>
`;

exports[`ConfirmEmailPage ionViewWillEnter on failed promise with non client error 1`] = `
<page-confirm-email
  clickMessage={[Function String]}
  linkText={[Function String]}
  navCtrl={[Function Object]}
  navParams={[Function NavParams]}
  thankYouMessage={[Function String]}
  toastProvider={[Function Object]}
  userProvider={[Function Object]}
>
  
  <draught-header />
  
  
  <ion-content
    padding=""
  >
    
    
    <ion-row>
      
      
      <ion-col
        class="below-header"
        col-12=""
        col-lg-6=""
        col-md-6=""
        col-sm-10=""
        col-xl-4=""
        offset-lg-3=""
        offset-md-3=""
        offset-sm-1=""
        offset-xl-4=""
        size="20px"
        text-justify=""
      >
        
        
        <p>
          Thank you for confirming your email!
        </p>
        
        
        <p>
          Please click the link below to sign in.
        </p>
        
        
        <div
          text-center=""
        >
          
          
          <a
            class="underline"
            text-center=""
          >
            Sign in.
          </a>
          
        
        </div>
        
      
      </ion-col>
      
    
    </ion-row>
    
  
  
  </ion-content>
</page-confirm-email>
`;

exports[`ConfirmEmailPage should initialize the ConfirmEmailPage 1`] = `
<page-confirm-email
  clickMessage={[Function String]}
  linkText={[Function String]}
  navCtrl={[Function Object]}
  navParams={[Function NavParams]}
  thankYouMessage={[Function String]}
  toastProvider={[Function Object]}
  userProvider={[Function Object]}
>
  
  <draught-header />
  
  
  <ion-content
    padding=""
  >
    
    
    <ion-row>
      
      
      <ion-col
        class="below-header"
        col-12=""
        col-lg-6=""
        col-md-6=""
        col-sm-10=""
        col-xl-4=""
        offset-lg-3=""
        offset-md-3=""
        offset-sm-1=""
        offset-xl-4=""
        size="20px"
        text-justify=""
      >
        
        
        <p>
          Thank you for confirming your email!
        </p>
        
        
        <p>
          Please click the link below to sign in.
        </p>
        
        
        <div
          text-center=""
        >
          
          
          <a
            class="underline"
            text-center=""
          >
            Sign in.
          </a>
          
        
        </div>
        
      
      </ion-col>
      
    
    </ion-row>
    
  
  
  </ion-content>
</page-confirm-email>
`;
